ambry-store/src/main/java/com.github.ambry.store/DiskSpaceAllocator.java

```diff
@@ -0,0 +1,335 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.store;
+
+import com.github.ambry.utils.Utils;
+import java.io.File;
+import java.io.FileFilter;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+class DiskSpaceAllocator {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DiskSpaceAllocator.class);
+  private static final String RESERVE_FILE_PREFIX = "reserve_";
+  private static final FileFilter RESERVE_FILE_FILTER =
+      pathname -> pathname.isFile() && pathname.getName().startsWith(RESERVE_FILE_PREFIX);
+  private static final FileFilter FILE_SIZE_DIR_FILTER = pathname -> {
+    try {
+      return pathname.isDirectory() && Long.parseLong(pathname.getName()) >= 0;
```

Comment:
No strong reason. I just wanted to make the parsing easier. I made the change to prefix it with  "reservedir_"